lnwallet/chancloser/chancloser.go

```diff
@@ -837,6 +873,175 @@ func (c *ChanCloser) proposeCloseSigned(fee btcutil.Amount) (*lnwire.ClosingSign
 	return closeSignedMsg, nil
 }
 
+// handleRemoteProposal sanity checks the remote's ClosingSigned message and
+// tries to determine an acceptable fee to reply with. It may return:
+// - a message and no error (continue negotiation)
+// - no message and an error (negotiation failed)
+// - no message and no error (indicating we agree with the remote's fee)
+func (c *ChanCloser) handleRemoteProposal(
+	remoteMsg *lnwire.ClosingSigned) (lnwire.Message, error) {
+
+	isFunder := c.cfg.Channel.IsInitiator()
+
+	// If FeeRange is set, perform FeeRange-specific checks.
+	if remoteMsg.FeeRange != nil {
+		// If legacyNegotiation is already set, fail outright.
+		if c.legacyNegotiation {
+			return nil, ErrCloseTypeChanged
+		}
+
+		// Set rangeNegotiation to true.
+		c.rangeNegotiation = true
+
+		// Get the intersection of our two FeeRanges if one exists.
+		overlap := c.idealFeeRange.GetOverlap(remoteMsg.FeeRange)
+
+		if isFunder {
+			// If the fundee replies with a FeeRange, there must be
+			// overlap with our FeeRange.
+			//
+			// BOLT#02:
+			// - otherwise (it is not the funder)
+			//   - ...
+			//   - otherwise
+			//     - MUST propose a fee_satoshis in the overlap
+			//	 between received and (about-to-be) sent
+			//       fee_range.
+			if overlap == nil {
+				return nil, ErrNoRangeOverlap
```

Comment:
What if instead we just have the funder send their desired fee rate? Based on inbound issues, people seem to want a trait where their co-op close actually _never_ fails, instead deferring to w/e the funder wanted since they're the one paying for fees.

Reply:
This is a spec-level check, if the fundee does not send an overlapping fee-range, we fail